src/GafferSceneUI/StandardLightVisualiser.cpp

```diff
@@ -996,73 +996,81 @@ IECoreGL::ConstRenderablePtr StandardLightVisualiser::quadPortal( const V2f &siz
 	p.push_back( V3f( size.x/2, size.y/2, 0 ) );
 	p.push_back( V3f( -size.x/2, size.y/2, 0 ) );
 
-	// 45 degree hatch outside the area (-1 / 1 space)
+	// 45 degree hatch outside the portal area (when centered at the origin)
 
 	// Space between the lines
-	static const float spacing = 0.05f;
-	// size.x of the shaded frame area
-	static const float fw = 0.25f;
+	const float spacing = 0.05f * visualiserScale;
```

Comment:
The opposite is bad too though, as at 100x, the lines are 0.05 real world units apart and it's just solid black... Maybe we just make the spacing a proportion of the shortest side.

Reply:
I don't follow. If I scale my light 100x (scaling is the only thing you can do in Gaffer at present, we don't have width/height controls) I get exactly the same number of lines as I did at 1x scale, and it all looks great. It's only when I mess with _visualiser_ scale that it all goes wrong.

Reply:
How about this as a compromise: https://github.com/GafferHQ/gaffer/pull/3616/commits/d6f5ef6cb5c0bd2c4c1fbaba8504afecba6954a5

In `StandardLightVisualiser` we keep the old behaviour (as we only have a fixed quad size, so people _have_ to scale the light). Then @danieldresser-ie can call it with `visualiserScale` for his lights, where they have `width` and `height`?

Reply:
> I don't follow. If I scale my light 100x (scaling is the only thing you can do in Gaffer at present, we don't have width/height controls) I get exactly the same number of lines as I did at 1x scale, and it all looks great. It's only when I mess with _visualiser_ scale that it all goes wrong.

Did you see Daniels comment earlier that it looks terrible at IE as they have big lights, so it's a wall of black hatching?

Reply:
> Did you see Daniels comment earlier that it looks terrible at IE as they have big lights, so it's a wall of black hatching?

Yes, I assumed that was your motivation for breaking the Cinesite hatching ;)

> How about this as a compromise: d6f5ef6

Seems OK, but I presume Daniel will end up doing something like `hatchingScale = 1 / max( width, height )`, in which case maybe we should just do that automatically internally? I presume you were thinking along similar lines when you said "Maybe we just make the spacing a proportion of the shortest side."

Reply:
I'd assumed he'll just do `hatchingScale = visualiserScale`, as that's what he wanted originally?

Reply:
I thought of one potential benefit of keeping it de-coupled from w/h - when we do support `width`/`height` editing, any interactive manipulations might be less distracting (though still not ideal) than with something based on the dimensions? I guess ideally, to make that the least eye melting it could be, we'd quantise the hatching to a grid rather than bottom left.

Reply:
> I'd assumed he'll just do hatchingScale = visualiserScale, as that's what he wanted originally?

But surely that only fixes the high hatching density for large width and/or height if the visualiser scale is adjusted to compensate?

> I thought of one potential benefit of keeping it de-coupled from w/h - when we do support width/height editing, any interactive manipulations might be less distracting (though still not ideal) than with something based on the dimensions? I guess ideally, to make that the least eye melting it could be, we'd quantise the hatching to a grid rather than bottom left.

Agreed. Perhaps it'd also be nice to transition between hatching densities by introducing new lines smoothly in between the existing ones. But I don't think any of this is helped by pushing a `scale` parameter into `quadPortal()` now - then all a client can do is make some poor approximation of the ideal behaviour, and we're a bit more constrained to implement that ideal behaviour later.

Reply:
> and we're a bit more constrained to implement that ideal behaviour later.

Actually, this is overstating things. I don't see any harm in a scale parameter if it helps Daniel in the short term.  d6f5ef6 will do for me...

Reply:
> But surely that only fixes the high hatching density for large width and/or height if the visualiser scale is adjusted to compensate?

Yeah, I guess related to the other point above though, when I was testing it as I was writing, there is something nice about looking at the scene with consistent hatch spacing across the lights in a scene. It bugs me that they scale with the light in stock Gaffer, but there's nothing we can do about that. 

One of the requests we had originally that made us decouple ornaments from geometry scaling was that uniform scale for common elements helps you understand depth better on a flat screen.  It felt like the vis scale linked approach helped with that, as no matter the size of your lights (at IE, for now) you'd be able to grok their relative position/sizes better.

I think for a scene where the lighter (or facility/show) has set a sensible visualiser scale for the working world scale, uniform hatching wins for me from that perspective. If you've not adjusted vis scale accordingly in the first, then so many other things look bad.

Im happy either way though if you and @danieldresser-ie want to work out what we want to go with.

Reply:
Thanks for the explanation. I think d6f5ef6 is the best we can do for now then - doesn't break the Gaffer visualisations, and allows the IELight visualisations to be a bit nicer. We can revisit when we start using width/height rather than scale in Gaffer.

Reply:
> It bugs me that they scale with the light in stock Gaffer, but there's nothing we can do about that.

Any reason we can't divide off the world space scale when determining the hatching?  Or do visualisers not get recomputed when they're scaled?  Hmm, it's too bad, making the hatching independent of the size of the light seems like the obviously correct answer, to keep all the lights nicely matching each other, and have everything adjust when a global visualiserScale for the scene is dialed in.

I guess I'm just lucky that I've already been trying to discourage anyone from scaling lights at IE.

Reply:
> Any reason we can't divide off the world space scale when determining the hatching? Or do visualisers not get recomputed when they're scaled?

Yep, visualisers are generated before scaling. There are a couple of options though, one of which Tom and I discussed yesterday :

- Use a shader to generate the hatching on the fly
- Use a custom `IECoreGL::Renderable` subclass to generate the hatching on the fly

We plan to revisit this at some point, but I don't think it's worth holding things up for at present.